Anime-App/Feature/Anime/ViewModel/AnimeViewModel.swift

```diff
@@ -8,48 +8,44 @@
 import Foundation
 
 class AnimeViewModel {
-    func getAnime(completion: @escaping (Result<[Anime], Error>) -> Void) {
+    
+    var showSpinner = false
+    
+    private(set) var animes: [Anime] = []
+    
+    func getAnime(completion: @escaping (Result<Void,NetworkingError>) -> Void) {
         // get request
-        guard let url = URL(string: "https://kitsu.io/api/edge/anime") else { return }
-        URLSession.shared.dataTask(with: url) { data, response, error in
-            guard error == nil else {
-                completion(.failure(error!))
-                return
-            }
-            do {
-                guard let data = data else {return}
-                let decoder = JSONDecoder()
-                decoder.keyDecodingStrategy = .convertFromSnakeCase
-                let anime = try decoder.decode(AnimeResult.self, from: data)
-                completion(.success(anime.data))
-                print(anime)
-            } catch {
-                completion(.failure(error))
+        let url = UrlEndpoint.baseUrl.endpoint + UrlEndpoint.getAnime.endpoint
+        
+        NetworkingManager.shared.request(url, type: AnimeResult.self) { res in
+            self.showSpinner = true
```

Comment:
If you have a flag here for showing the spinner do you even need the isrefreshing in the networking manager? Cos now you're doing it in two different places

This is messy Tobi...You need to cut this bad habit out...

Reply:
Can you review this in this file and also as well look at other places to make sure you're not doing the same thing